docs/developer/buddypress-database-diagram.md

```diff
@@ -0,0 +1,388 @@
+  - `secondary_item_id`: Additional context ID, if applicable.
+  - `date_recorded`: When the activity was recorded.
+  - `hide_sitewide`: Whether the activity is hidden from the sitewide activity stream.
+  - `mptt_left` & `mptt_right`: Used for threaded comments/activities.
+  - `is_spam`: Marks the activity as spam.
+
+#### Example Query:
+
+```sql
+SELECT * FROM wp_bp_activity WHERE user_id = 1;
+```
+
+### wp_bp_activity_meta
+
+Stores metadata associated with activity entries.
+
+- **Columns:**
+  - `id`: Meta ID.
+  - `activity_id`: The ID of the associated activity.
+  - `meta_key`: Metadata key.
+  - `meta_value`: Metadata value (usually stored as serialized data).
+
+---
+
+## Friends Tables
+
+### wp_bp_friends
+
+This table manages the friendship connections between users.
+
+- **Columns:**
+  - `id`: Unique ID for each friendship connection.
+  - `initiator_user_id`: The user who sent the friend request.
+  - `friend_user_id`: The user who received the friend request.
+  - `is_confirmed`: Whether the friendship has been confirmed (1) or is pending (0).
+  - `is_limited`: Whether the friendship has any limitations.
+  - `date_created`: Timestamp of when the friendship was initiated.
+
+#### Example Query:
+
+```sql
+SELECT * FROM wp_bp_friends WHERE initiator_user_id = 1;
+```
+
+---
+
+## Groups Tables
+
+### wp_bp_groups
+
+This table stores the details of groups created in BuddyPress.
+
+- **Columns:**
+  - `id`: Unique ID for the group.
+  - `creator_id`: User ID of the group creator.
+  - `name`: Name of the group.
+  - `slug`: Unique slug for the group.
+  - `description`: Group description.
+  - `status`: Group visibility status (public, private, hidden).
+  - `parent_id`: Parent group ID (if applicable).
+  - `enable_forum`: Whether the group has forums enabled.
+  - `date_created`: When the group was created.
+
+### wp_bp_groups_groupmeta
+
+Stores metadata associated with groups.
+
+- **Columns:**
+  - `id`: Unique ID for group metadata.
+  - `group_id`: The ID of the associated group.
+  - `meta_key`: Metadata key.
+  - `meta_value`: Metadata value (usually stored as serialized data).
+
+### wp_bp_groups_members
+
+Manages membership of users in groups.
+
+- **Columns:**
+  - `id`: Unique membership ID.
+  - `group_id`: The ID of the group.
+  - `user_id`: The ID of the user.
+  - `inviter_id`: The ID of the user who invited the member (if applicable).
+  - `is_admin`: Whether the user is an admin of the group.
+  - `is_mod`: Whether the user is a moderator of the group.
+  - `user_title`: The role or title of the user within the group.
+  - `date_modified`: When the membership was last modified.
+  - `is_confirmed`: Whether the membership is confirmed.
+  - `is_banned`: Whether the user is banned from the group.
+  - `invite_sent`: Whether an invitation has been sent.
+
+#### Example Query:
+
+```sql
+SELECT * FROM wp_bp_groups_members WHERE group_id = 2;
+```
+
+---
```

Comment:
It should be in `docs/developer/components/groups/db-schema.md` and use a table like the what I've written for the Activity above.